sdk/python/feast/infra/offline_stores/redshift.py

```diff
@@ -264,6 +264,11 @@ def to_arrow(self) -> pa.Table:
 
     def to_s3(self) -> str:
```

Comment:
What does the `        with self._query_generator() as query:` do? Should the upload be in that block, or is it fine outside?

Reply:
Should be ok since this is called within the self.to_df() call.